joomla/joomla-cms
# [Actionslog] New option to enable User Actions Log + Events To Log improvement

Pull Request for Issue #22627.

### Summary of Changes
- Add new option to enable User Actions Log. Disabled by default.
- Remove default value for `loggable_extensions` (Fix #22627 UX issue)
- Add a "Select All/None" for loggeable extensions to help selecting all or clearing all on one click.
- Add two strings for `actions_logging` new option to be reviewed.
```
COM_ACTIONLOGS_ACTIONS_LOGGING_DESC="Enable log of actions performed by users on your website for the selected events to be logged."
COM_ACTIONLOGS_ACTIONS_LOGGING_LABEL="Enable User Actions Log"

```

Disabled
![capture d ecran 2018-10-16 a 19 04 20](https://user-images.githubusercontent.com/2385058/47033945-59ef4800-d176-11e8-831e-8d5912d17651.png)

Enabled
![capture d ecran 2018-10-16 a 19 04 05](https://user-images.githubusercontent.com/2385058/47033964-670c3700-d176-11e8-8d41-0d12e47e0429.png)



### Testing Instructions
- Check issue #22627 for `loggable_extensions` UX changes.
- Test actions logging options (enabled/disabled), all selected, some selected, none selected.

Note: i don't have removed default core extensions to be logged, from database install/update, as it make sense to keep this default setting when first opening options. No more issue as User Actions Log is now disabled, by default.
